ref/vk/r_speeds.c

```diff
@@ -706,35 +711,98 @@ static const char *getMetricTypeName(r_speeds_metric_type_t type) {
 	return "UNKNOWN";
 }
 
-// Ideally, we'd just autocomplete the r_speeds_graphs cvar/cmd.
-// However, autocompletion is not exposed to the renderer. It is completely internal to the engine, see con_utils.c, var cmd_list.
-static void listMetrics( void ) {
-	if (gEngine.Cmd_Argc() > 1) {
-		Q_strncpy(g_speeds.frame.list_metrics_filter, gEngine.Cmd_Argv(1), sizeof(g_speeds.frame.list_metrics_filter));
-	} else {
-		g_speeds.frame.list_metrics_filter[0] = '\0';
+// Returns pointer to filename in filepath string.
+// Maybe function like this should be inside filesystem?
+// Examples:
+// on Windows: C:\Users\User\xash3d-fwgs\ref\vk\vk_rtx.c  ->  vk_rtx.c
+// on Linux:   /home/user/xash3d-fwgs/ref/vk/vk_rtx.c     ->  vk.rtx.c (imaginary example, not tested)
+static const char *get_filename_from_filepath( const char *filepath ) {
+	size_t length = Q_strlen( filepath );
+	int cursor = length - 1;
+	while ( cursor > 0 ) {
+		char c = filepath[cursor];
+		if ( c == '/' || c == '\\' ) {
+			// Advance by 1 char to skip the folder delimiter symbol itself, but
+			// make sure that we are not exceeding the length.
+			if ( cursor < length )
+				cursor += 1;
+
+			return &filepath[cursor];
+		}
+		cursor -= 1;
 	}
 
-	g_speeds.frame.list_metrics = true;
+	return NULL;
 }
 
-static void doListMetrics( void ) {
-	if (!g_speeds.frame.list_metrics)
+// Actually does the job of `r_speeds_mlist` and `r_speeds_mtable` commands.
+// We can't just directly call this function from little command handler ones, because
+// all the metrics calculations happen inside `R_SpeedsDisplayMore` function.
+static void doPrintMetrics( void ) {
+	if ( g_speeds.frame.metrics_print_mode == kSpeedsMprintNone )
 		return;
 
-	g_speeds.frame.list_metrics = false;
-	const char *const filter = g_speeds.frame.list_metrics_filter;
+	const char *header_format = NULL;
+	const char *line_format = NULL;
+	const char *row_format = NULL;
+	char line[64];
+	if ( g_speeds.frame.metrics_print_mode == kSpeedsMprintTable ) {
+		// Note:
+		// This table alignment method relies on monospace font
+		// and will have its alignment completly broken without one.
+		// That is why there is now cvar called `r_speeds_metrics_as_table`,
+		// which can toggle this printing method on/off.
+		header_format = "  | %-38s | %-10s | %-40s | %21s\n";
+		line_format   = "  | %.38s | %.10s | %.40s | %.21s\n";
+		row_format    = "  | ^2%-38s^7 | ^3%-10s^7 | ^5%-40s^7 | ^6%s:%d^7\n";
+		memset( line, '-', sizeof( line ) );
```

Comment:
Нуль-терминировать бы её, чтобы не устроить _хлопок_, если в `line_format` приедет ширина больше 63 в некотором будущем, когда все уже забудут, что тут происходит.
```suggestion
		memset( line, '-', sizeof( line ) );
		line[sizeof(line)-1] = '\0';
```

Вообще, я бы эту линию как-то по-другому рисовал, но хороших идей кроме того, что тупо руками эту строчку сделать статичной константой (ширины всё равно фиксированы), нет, пока пусть будет так.